components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/dao/SAMLSSOServiceProviderDAO.java

```diff
@@ -301,6 +209,103 @@ public boolean addServiceProvider(SAMLSSOServiceProviderDO serviceProviderDO)
+        resource.setProperty(IdentityRegistryResources.PROP_SAML_SSO_ASSERTION_CONS_URLS,
+                serviceProviderDO.getAssertionConsumerUrlList());
+        resource.addProperty(IdentityRegistryResources.PROP_DEFAULT_SAML_SSO_ASSERTION_CONS_URL,
+                serviceProviderDO.getDefaultAssertionConsumerUrl());
+        resource.addProperty(IdentityRegistryResources.PROP_SAML_SSO_ISSUER_CERT_ALIAS,
+                serviceProviderDO.getCertAlias());
+        resource.addProperty(IdentityRegistryResources.PROP_SAML_SSO_LOGIN_PAGE_URL,
+                serviceProviderDO.getLoginPageURL());
+        resource.addProperty(
+                IdentityRegistryResources.PROP_SAML_SSO_NAMEID_FORMAT,
+                serviceProviderDO.getNameIDFormat());
+        resource.addProperty(IdentityRegistryResources.PROP_SAML_SSO_SIGNING_ALGORITHM, serviceProviderDO
+                .getSigningAlgorithmUri());
+        resource.addProperty(IdentityRegistryResources.PROP_SAML_SSO_DIGEST_ALGORITHM, serviceProviderDO
+                .getDigestAlgorithmUri());
+        if (serviceProviderDO.getNameIdClaimUri() != null
+                && serviceProviderDO.getNameIdClaimUri().trim().length() > 0) {
+            resource.addProperty(
+                    IdentityRegistryResources.PROP_SAML_SSO_ENABLE_NAMEID_CLAIMURI,
+                    "true");
+            resource.addProperty(
+                    IdentityRegistryResources.PROP_SAML_SSO_NAMEID_CLAIMURI,
+                    serviceProviderDO.getNameIdClaimUri());
+        } else {
+            resource.addProperty(
+                    IdentityRegistryResources.PROP_SAML_SSO_ENABLE_NAMEID_CLAIMURI,
+                    "false");
+        }
+
+        String doSingleLogout = serviceProviderDO.isDoSingleLogout() ? "true" : "false";
+        resource.addProperty(IdentityRegistryResources.PROP_SAML_SSO_DO_SINGLE_LOGOUT, doSingleLogout);
+        if (serviceProviderDO.isDoSingleLogout()) {
+            if (StringUtils.isNotBlank(serviceProviderDO.getSloResponseURL())) {
+                resource.addProperty(IdentityRegistryResources.PROP_SAML_SLO_RESPONSE_URL,
+                        serviceProviderDO.getSloResponseURL());
+            }
+            if (StringUtils.isNotBlank(serviceProviderDO.getSloRequestURL())) {
+                resource.addProperty(IdentityRegistryResources.PROP_SAML_SLO_REQUEST_URL,
+                        serviceProviderDO.getSloRequestURL());
+            }
+        }
+
+        String doSignResponse = serviceProviderDO.isDoSignResponse() ? "true" : "false";
+        resource.addProperty(IdentityRegistryResources.PROP_SAML_SSO_DO_SIGN_RESPONSE,
+                doSignResponse);
+        String doSignAssertions = serviceProviderDO.isDoSignAssertions() ? "true" : "false";
+        resource.addProperty(IdentityRegistryResources.PROP_SAML_SSO_DO_SIGN_ASSERTIONS,
+                doSignAssertions);
+        if (serviceProviderDO.getRequestedClaimsList() != null
```

Comment:
Can't we replace this with CollectionUtils.isNotEmpty() [method](https://commons.apache.org/proper/commons-collections/apidocs/org/apache/commons/collections4/CollectionUtils.html#isNotEmpty(java.util.Collection)) in all the places where we check whether a list is null or empty?